src/adapter/src/coord.rs

```diff
@@ -467,6 +467,7 @@ pub struct PeekStageLinearizeTimestamp {
     validity: PlanValidity,
     plan: mz_sql::plan::SelectPlan,
     max_query_result_size: Option<u64>,
+    max_query_heap_size: Option<u64>,
```

Comment:
Seems like a good idea to bundle these into a `QueryResourceLimits` or similar, to making passing them around less boilerplate-y.

Reply:
I think there's maybe a larger refactoring around here (the first ~900 lines of `coord.rs` are types without documentation, other than some field-level doccomments). Imo, maybe this PR isn't the place to do that, but rather a more holistic sweep?

Reply:
Having said that, the next several pages of diffs do seem to be more boilerplate, pairing up the new parameter with the prior one. I may be coming around!

Reply:
Should have called this a nit! Imo it's good to follow the boy scout rule when it's not too much of a hassle, because we might never get to that holistic refactor. But nothing I'd block the PR on.

Reply:
+1 to bundling into a `QueryResourceLimits` but also not blocking